src/framework/smoke_job.hpp

```diff
@@ -287,25 +287,25 @@ class SmokeJob {
       }
 
       // Get priority.
-      size_t optionPriority = 1;
+      size_t optionShares = 1;
 
-      Result<JSON::Number> _priority =
+      Result<JSON::Number> _shares =
         json.get().find<JSON::Number>(
-          "tasks[" + stringify(i) +"].totalTasks");
+          "tasks[" + stringify(i) +"].shares");
 
-      if (_priority.isSome()) {
-        optionPriority = _priority.get().as<size_t>();
+      if (_shares.isSome()) {
+        optionShares = _shares.get().as<size_t>();
       }
 
-      jobs.push_back(
+      jobs.push_back(std::make_shared<SmokeJob>(
         SmokeJob(i,
                  optionCommand.get().value,
                  optionResources.get(),
                  optionTotalTasks,
                  optionTargetHostname,
                  optionUri,
-                 optionPriority));
-      jobs.back().print();
+                 optionShares)));
+      jobs.back()->print();
```

Comment:
Could we override ```std::ostream& operator << (std::ostream &strm, const QueueBlock &a)``` instead of print method?
Then, we could use ```LOG(INFO) << jobs.back();```